osu.Game/Rulesets/Difficulty/IDifficultyAttributes.cs

```diff
@@ -49,42 +42,40 @@ public class DifficultyAttributes
         public int MaxCombo { get; set; }
 
         /// <summary>
-        /// Creates new <see cref="DifficultyAttributes"/>.
-        /// </summary>
-        public DifficultyAttributes()
-        {
-        }
-
-        /// <summary>
-        /// Creates new <see cref="DifficultyAttributes"/>.
-        /// </summary>
-        /// <param name="mods">The mods which were applied to the beatmap.</param>
-        /// <param name="starRating">The combined star rating of all skills.</param>
-        public DifficultyAttributes(Mod[] mods, double starRating)
-        {
-            Mods = mods;
-            StarRating = starRating;
-        }
-
-        /// <summary>
-        /// Converts this <see cref="DifficultyAttributes"/> to osu-web compatible database attribute mappings.
+        /// Converts this <see cref="IDifficultyAttributes"/> to osu-web compatible database attribute mappings.
         /// </summary>
         /// <remarks>
         /// See: osu_difficulty_attribs table.
         /// </remarks>
-        public virtual IEnumerable<(int attributeId, object value)> ToDatabaseAttributes()
-        {
-            yield return (ATTRIB_ID_MAX_COMBO, MaxCombo);
-        }
+        public IEnumerable<(int attributeId, object value)> ToDatabaseAttributes();
 
         /// <summary>
-        /// Reads osu-web database attribute mappings into this <see cref="DifficultyAttributes"/> object.
+        /// Reads osu-web database attribute mappings into this <see cref="IDifficultyAttributes"/> object.
         /// </summary>
         /// <param name="values">The attribute mappings.</param>
         /// <param name="onlineInfo">The <see cref="IBeatmapOnlineInfo"/> where more information about the beatmap may be extracted from (such as AR/CS/OD/etc).</param>
-        public virtual void FromDatabaseAttributes(IReadOnlyDictionary<int, double> values, IBeatmapOnlineInfo onlineInfo)
+        public void FromDatabaseAttributes(IReadOnlyDictionary<int, double> values, IBeatmapOnlineInfo onlineInfo);
+    }
+
+    /// <summary>
+    /// Represents a full, minimal implementation of <see cref="IDifficultyAttributes"/>.
+    /// </summary>
+    public class EmptyDifficultyAttributes : IDifficultyAttributes
```

Comment:
Needs to be renamed. "Empty" difficulty attributes suggests no difficulty attributes. And yet there are two.

Reply:
Hmm, any suggestions? "Empty" seems logical to me because it adds no ruleset-specific attributes and only the two generic ones exist.

Reply:
I'd be ok with dropping the `Empty` from that yeah.